streams/src/main/java/org/apache/kafka/streams/state/internals/InMemoryKeyValueStore.java

```diff
@@ -103,6 +105,19 @@ public void putAll(final List<KeyValue<Bytes, byte[]>> entries) {
         }
     }
 
+    @Override
+    public <PS extends Serializer<P>, P> KeyValueIterator<Bytes, byte[]> prefixScan(final P prefix, final PS prefixKeySerializer) {
+        Objects.requireNonNull(prefix, "prefix cannot be null");
+        Objects.requireNonNull(prefixKeySerializer, "prefixKeySerializer cannot be null");
+
+        final Bytes from = Bytes.wrap(prefixKeySerializer.serialize(null, prefix));
+        final Bytes to = Bytes.increment(from);
+
+        return new DelegatingPeekingKeyValueIterator<>(
+            name,
+            new InMemoryKeyValueIterator(map.subMap(from, true, to, false).keySet(), true));
```

Comment:
nit: The last parenthesis should go to a new line. I know that we are not consistent throughout the code base (that is why this comment is prefixed with "nit") but that is actually the code style, we agreed upon. If you need to push another commit you can fix this, otherwise it's fine. 
```suggestion
        return new DelegatingPeekingKeyValueIterator<>(
            name,
            new InMemoryKeyValueIterator(map.subMap(from, true, to, false).keySet(), true)
        );
```